.test-infra/jenkins/metrics_report/generate_report.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This script will be run by Jenkins as a Metric Report.
+# Versus the tip of the target branch
+# Parameters
+#  $1: INFLUX_DB
+#  $2: INFLUX_HOST
+#  $3: INFLUX_PORT
+#
+
+set -e
+set -v
+
+VENV_DIR="metrics_report/venv"
+
+INFLUX_DB=$1
+INFLUX_HOST=$2
+INFLUX_PORT=$3
+
+echo "INFLUX_DB = ${INFLUX_DB}"
+echo "INFLUX_HOST = ${INFLUX_HOST}"
+echo "INFLUX_PORT = ${INFLUX_PORT}"
+
+python3 -m venv "${VENV_DIR}"
```

Comment:
What Python version will be used here? 3.5? 3.8? It's hard to say. How about creating a dedicated tox suite? We have got many tox suites for different kind of tools: sphinx, autoformatter, mypy, linters. The advantages are: 

- you can specify what Python version you want to use
- tox takes care of creating virtualenvs and installing deps
- everything could be found in one place

See https://github.com/apache/beam/blob/master/sdks/python/tox.ini for examples

Reply:
I created `tox.ini`
```
[tox]
skipsdist = True
envlist = py37-test,py37-run

[testenv:py37-test]
deps = -r requirements.txt
commands =
    python --version
    python -m unittest dashboards_parser.py

[testenv:py37-generate-report]
deps = -r requirements.txt
commands =
    python --version
    python report_generator.py {posargs}
```
however tox seems to be not available by default on Jenkins node. 
This could would run on any version python 3. I saw this vritualenv creating pattern in other bash scripts. WDYT of keeping this way for sake of simplicity?